app/controllers/katello/api/v2/errata_controller.rb

```diff
@@ -3,21 +3,28 @@ class Api::V2::ErrataController < Api::V2::ApiController
     apipie_concern_subst(:a_resource => N_("an erratum"), :resource => "errata")
     include Katello::Concerns::Api::V2::RepositoryContentController
 
+    before_action :find_host, :only => :index
+
     api :GET, "/errata", N_("List errata")
-    param :organization_id, :number, :desc => N_("organization identifier")
-    param :content_view_version_id, :number, :desc => N_("content view version identifier")
-    param :content_view_filter_id, :number, :desc => N_("content view filter identifier")
-    param :repository_id, :number, :desc => N_("repository identifier")
-    param :environment_id, :number, :desc => N_("environment identifier")
+    param :organization_id, :number, :desc => N_("Organization identifier")
+    param :content_view_version_id, :number, :desc => N_("Content View Version identifier")
+    param :content_view_filter_id, :number, :desc => N_("Content View Filter identifier")
+    param :repository_id, :number, :desc => N_("Repository identifier")
+    param :environment_id, :number, :desc => N_("Environment identifier")
     param :cve, String, :desc => N_("CVE identifier")
-    param :errata_restrict_applicable, :bool, :desc => N_("show only errata with one or more applicable hosts")
-    param :errata_restrict_installable, :bool, :desc => N_("show only errata with one or more installable hosts")
+    param :host_id, :number, :desc => N_("Host id to list applicable errata for")
+    param :errata_restrict_applicable, :bool, :desc => N_("Return errata that are applicable to one or more hosts (defaults to true if host_id is specified)")
+    param :errata_restrict_installable, :bool, :desc => N_("Return errata that are upgradable on one or more hosts")
+    param :available_for, String, :desc => N_("Return errata that can be added to the specified object.  The values 'content_view_version' and 'content_view_filter are supported.")
     param_group :search, Api::V2::ApiController
     def index
-      params[:errata_restrict_applicable] = false if ::Foreman::Cast.to_bool(params[:errata_restrict_installable])
       super
     end
 
+    def available_for_content_view_version(version)
+      version.available_errata
+    end
```

Comment:
I don't see this method used anywhere, nor do i see the available_for=content_view_version actually working properly.

Reply:
It is called from https://github.com/Katello/katello/pull/7324/commits/1c9cfb8d87fe2441d5ca1b22be18a38e543d92b6#diff-afb34388d8bbd87b85d8adbd9d0b2cb1R69

I started reworking this PR per https://github.com/Katello/katello/pull/7324#issuecomment-386642181 but figured I should wait for #7324 to be merged before finishing this up, so this was incomplete.

I'll rebase this on the current master shortly.